src/_pytest/outcomes.py

```diff
@@ -157,8 +157,11 @@ def skip(
         The message to show the user as reason for the skip.
 
     :param allow_module_level:
-        Allows this function to be called at module level, skipping the rest
-        of the module. Defaults to False.
+        Allows this function to be called at module level.
+        Raising the skip exception in a module will stop the execution of the module.
+        Additionally, the module will no longer be collected.
```

Comment:
For someone not familiar with pytest internals, it’s still not obvious that none of the tests defined before the skip will ever be run or reported.

Reply:
thanks for the feedback, reworded it to explicitly state that it prevents collection